tests/runner/src/env.rs

```diff
@@ -693,8 +780,28 @@ impl GreptimeDB {
 impl Database for GreptimeDB {
     async fn query(&self, ctx: QueryContext, query: String) -> Box<dyn Display> {
         if ctx.context.contains_key("restart") && self.env.server_addrs.server_addr.is_none() {
-            self.env.restart_server(self).await;
+            self.env.restart_server(self, false).await;
+        } else if let Some(version) = ctx.context.get("version") {
+            if self.env.old_bins_dir.lock().unwrap().is_none() {
+                // save old bins dir
+                *self.env.old_bins_dir.lock().unwrap() = self.env.bins_dir.lock().unwrap().clone();
```

Comment:
Seems `old_bins_dir` is always latest bins dir. I suggest using a hashmap to manage bins dirs across versions, to make codes more understandable.